lib/oban.ex

```diff
@@ -840,4 +861,13 @@ defmodule Oban do
 
     with {:error, error} <- conf.engine.init(conf, opts), do: raise(error)
   end
+
+  defp validate_queue_exists!(name, opts) do
+    queue = opts[:queue]
+    local_only = opts[:local_only]
+
+    if local_only && is_nil(Registry.whereis(name, {:producer, to_string(queue)})) do
```

Comment:
These are both booleans, the `and` version should work:

```suggestion
    if local_only and is_nil(Registry.whereis(name, {:producer, to_string(queue)})) do
```

Reply:
Wait, Isn't it optional to pass this in? In that case we should at least default the opt to false right?

That's why I did the &&

Reply:
Yes, my mistake. This is still hit if `local_only` is `nil`.